### PR TITLE
Added File IO in Perl

### LINKED ISSUE #1372: Add Prime Number in Java
See requirements: https://sample-programs.therenegadecoder.com/projects/prime-number/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/j/java

### LINKED ISSUE #1510: Add File IO in Perl
See requirements: https://sample-programs.therenegadecoder.com/projects/file-io/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/p/perl

### PR DESCRIPTION
**Pull request by EwertonQueiroz**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #1510

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2]) 

### Documentation

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [ ] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [ ] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [ ] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [ ] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

File: archive/j/java/PrimeNumber.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+public class PrimeNumber {
+
+	public static void main(String[] args) throws Exception {
+		try {
+			int n = Integer.parseInt(args[0]);
+			
+			if (n < 0)
```
Comment (jrg94) [inline archive/j/java/PrimeNumber.java]: I'm not in love with the way this exception is handled. To me, it seems like it makes more sense to have some sort of branch that checks this initially before printing the error. That said, this passes all of our tests, so it's no big deal.

Comment (EwertonQueiroz) [reply archive/j/java/PrimeNumber.java]: I guess that I misunderstood you, what do you want is to check the parameter args[0] before assignment?

**Comment by EwertonQueiroz** [comment]
> Nice work! Can you add a reference to this solution to the README?

See #1654

**Comment by jrg94** [comment]
You're going to need to update your branch with the changes on Master. We recently changed the way we do testing, so the tests are failing due to that.

**Pull request merged**